Tiltfile

```diff
@@ -388,6 +392,104 @@ def prepare_all():
     )
     local(cmd, env = settings.get("kustomize_substitutions", {}))
 
+# create cluster template resources from cluster-template files in the templates directory
+def cluster_templates():
+    substitutions = settings.get("kustomize_substitutions", {})
+
+    template_dirs = settings.get("template_dirs", {
+        "docker": ["./test/infrastructure/docker/templates"],
+    })
+
+    for provider, provider_dirs in template_dirs.items():
+        for template_dir in provider_dirs:
+            template_list = [filename for filename in listdir(template_dir) if os.path.basename(filename).endswith("yaml")]
+            for filename in template_list:
+                deploy_templates(filename, provider, substitutions)
+
+def deploy_templates(filename, provider, substitutions):
+    # validate filename exists
+    if not os.path.exists(filename):
+        fail(filename + " not found")
+
+    os.environ["NAMESPACE"] = substitutions.get("NAMESPACE", "default")
+    os.environ["KUBERNETES_VERSION"] = substitutions.get("KUBERNETES_VERSION", "v1.24.0")
+    os.environ["CONTROL_PLANE_MACHINE_COUNT"] = substitutions.get("CONTROL_PLANE_MACHINE_COUNT", "1")
+    os.environ["WORKER_MACHINE_COUNT"] = substitutions.get("WORKER_MACHINE_COUNT", "3")
+
+    basename = os.path.basename(filename)
+    if basename.endswith(".yaml"):
+        if basename.startswith("clusterclass-"):
+            template_name = basename.replace("clusterclass-", "").replace(".yaml", "")
+            deploy_clusterclass(template_name, provider, filename)
+        elif basename.startswith("cluster-template-"):
+            template_name = basename.replace("cluster-template-", "").replace(".yaml", "")
+            deploy_cluster_template(template_name, provider, filename)
+
+def deploy_clusterclass(template_name, provider, filename):
+    apply_clusterclass_cmd = "cat " + filename + " | " + envsubst_cmd + " | " + kubectl_cmd + " apply -f - && echo \"ClusterClass created from\'" + filename + "\', don't forget to delete\n\""
+    delete_clusterclass_cmd = kubectl_cmd + " delete clusterclass " + template_name + ' --ignore-not-found=true; echo "\n"'
+
+    local_resource(
+        name = template_name,
+        cmd = apply_clusterclass_cmd,
```

Comment:
Isn't it a problem that we have the cmd here without sh -c? (same for cluster template)

Reply:
It should be fine. The [docs](https://docs.tilt.dev/api.html#api.local_resource) on this function mention if it's a string, it will be executed with "sh -c" automatically.

Reply:
Ah fun. Interesting that local_resource and cmd_button have different fields with different semantics